dockstore-client/src/main/java/io/dockstore/client/cli/nested/ApiClientExtended.java

```diff
@@ -236,6 +240,43 @@ public void createBodyPart(MultiPart multiPart, String key, Object formObject) {
                     buildResponseHeaders(response),
                     respBody);
             }
+        } catch (ProcessingException ex) {
+            // This could be caused by a failed Jersey Interceptor/filter, missing message body writers, or other IO exceptions.
+            // Mainly, this error is thrown when the provided WES URL is invalid. For more details, see:
+            // https://docs.oracle.com/javaee/7/api/index.html?javax/ws/rs/ProcessingException.html
+            errorMessage(MessageFormat.format("There was an error processing the HTTP request. This could be caused by an invalid URL: {0}",
+                ex.getMessage()), Client.CONNECTION_ERROR);
+            return null;
+        } catch (ApiException ex) {
+            // Different WES servers provide error messages with different levels of verbosity/usefulness, so include both a default
+            // message and the message provided from the WES server in the printed error.
+            switch (ex.getCode()) {
+            case HttpStatus.SC_BAD_REQUEST:
+                errorMessage(MessageFormat.format("[{0}] The WES request was malformed: {1}",
+                    ex.getCode(), ex.getMessage()), Client.API_ERROR);
+                break;
+            case HttpStatus.SC_UNAUTHORIZED:
+                errorMessage(MessageFormat.format("[{0}] The WES request is unauthorized to be performed on the target WES server: {1}",
+                    ex.getCode(), ex.getMessage()), Client.API_ERROR);
+                break;
+            case HttpStatus.SC_FORBIDDEN:
+                errorMessage(MessageFormat.format("[{0}] The provided credentials are not authorized to make this WES request: {1}",
+                    ex.getCode(), ex.getMessage()), Client.API_ERROR);
+                break;
+            case HttpStatus.SC_NOT_FOUND:
+                errorMessage(MessageFormat.format("[{0}] The WES server was unable to locate the target entry: {1}",
```

Comment:
Hmm, this could be confusing -- if the CLI invokes an endpoint at `https://<wes-server>/nonexistent/path`, you'll probably get a 404, and the beginning of this message may imply something else.

Reply:
You are correct, the error message changes based on how incorrect the URL is, some exampels:

---
 1.
```
[WES]
url: https://12345.execute-api.us-west-2.amazonaws.com/prod/ga4gh/wes/v1/fake
```
returns a response of:
```
12:28:36.294 [main] ERROR io.dockstore.client.cli.ArgumentUtility - [404] The WES server was unable to locate the target entry: <long based 64 encoded string>
```

---
2.
```
[WES]
url: https://12345.execute-api.us-west-2.amazonaws.com/fake/prod/ga4gh/wes/v1
```
returns a response of:
```
12:34:21.544 [main] ERROR io.dockstore.client.cli.ArgumentUtility - [403] The provided credentials are not authorized to make this WES request: {"message":"Forbidden"}
```

---
3.
```
[WES]
url: notAServerAtAll
```
returns a response of:
```
12:35:32.086 [main] ERROR io.dockstore.client.cli.ArgumentUtility - There was an error processing the HTTP request. This could be caused by an invalid URL: value is null
```

---

So we have a decent variety of errors all related to invalid URLs. (Of the above, only 1 is a 404). The main reason I added the "The WES server was unable to locate target entry" was due to the fact that all 404 responses specified in the WES schema are when a workflow ID is invalid. Perhaps this could be modified to "Invalid WES endpoint"? Not sure, thoughts?

Reply:
What is a "target entry"? I see it in the AGC message, but I have no idea what it means. I was first parsing it with Dockstore eyes, and thinking "entry" meant tool/workflow/descriptor, but I don't think that's it.

You could use the WES standard response message, "The requested workflow run wasn't found", but that would only apply in some cases.